packages/fuels-programs/src/calls/call.rs

```diff
@@ -0,0 +1,76 @@
+use fuel_tx::{Output, Receipt};
+use fuel_types::{Address, AssetId};
+use fuel_vm::fuel_asm::PanicReason;
+
+use fuels_core::constants::FAILED_TRANSFER_TO_ADDRESS_SIGNAL;
+use fuels_types::bech32::Bech32ContractId;
+
+use crate::logs::LogDecoder;
+use crate::{calls::contract_call::ContractCall, calls::script_call::ScriptCall};
+
+// Trait implemented by contract instances so that
+// they can be passed to the `set_contracts` method
+pub trait SettableContract {
+    fn id(&self) -> Bech32ContractId;
+    fn log_decoder(&self) -> LogDecoder;
+}
+
+pub(crate) trait ProgramCall {
```

Comment:
I love to see deduplication any day -- but :D 

I'm not seeing that we're using this new trait `ProgramCall` anywhere. Besides not having a use for it rn, it seems forced/unfortunately named. A ProgramCall which you cannot call -- the methods inside revolve mostly around transaction inputs/outputs. 

The dedupe effort can still be used without the trait by just calling the macro to produce the code. You're still in the same crate and you can have multiple impl blocks for a struct:
```rust
struct SomeStruct;
impl SomeStruct { 
    fn foo() {
        
    }
}
impl SomeStruct {
    fn goo() {
        
    }
}
```

also, you can then leave the macro here and call it below the original impl blocks in both call handlers, to keep the impls close.

Reply:
That makes a lot of sense, thanks a lot. The trait lead me to having a macro and now I can remove the trait haha.